src/common/tts/edge-tts.ts

```diff
@@ -0,0 +1,291 @@
+import { SpeakOptions } from './types'
+import { v4 as uuidv4 } from 'uuid'
+
+function mkssml(text: string, voice: string, rate: string, volume: string) {
+    return (
+        "<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' xml:lang='en-US'>" +
+        `<voice name='${voice}'><prosody pitch='+0Hz' rate='${rate}' volume='${volume}'>` +
+        `${text}</prosody></voice></speak>`
+    )
+}
+
+function ssmlHeadersPlusData(requestId: string, timestamp: string, ssml: string) {
+    return (
+        `X-RequestId:${requestId}\r\n` +
+        'Content-Type:application/ssml+xml\r\n' +
+        `X-Timestamp:${timestamp}Z\r\n` + // This is not a mistake, Microsoft Edge bug.
+        `Path:ssml\r\n\r\n` +
+        `${ssml}`
+    )
+}
+
+function getHeadersAndData(data: string) {
+    const headers: { [key: string]: string } = {}
+    data.slice(0, data.indexOf('\r\n\r\n'))
+        .split('\r\n')
+        .forEach((line) => {
+            const [key, value] = line.split(':', 2)
+            headers[key] = value
+        })
+    return { headers, data: data.slice(data.indexOf('\r\n\r\n') + 4) }
+}
+
+const trustedClientToken = '6A5AA1D4EAFF4E9FB37E23D68491D6F4'
+const wssURL =
+    'wss://speech.platform.bing.com/consumer/speech/synthesize/' +
```

Comment:
Should we add this host as one of the `host_permissions` in `manifest.json` and `manifest.firefox.json`?

Reply:
Not sure, but works well without it (in chrome).
Add it first.